smartsim/_core/launcher/dragon/dragonBackend.py

```diff
@@ -155,150 +347,314 @@ def _allocate_step(
             return to_allocate
 
     def _get_new_id(self) -> str:
-        with self._step_id_lock:
-            self._step_id += 1
-            return create_short_id_str() + "-" + str(self._step_id)
+        step_id = create_short_id_str() + "-" + str(self._step_id)
```

Comment:
Is the `DragonBackend` the place where a `step_id` should be located? Is there a `DragonStep` that could do it? 

Do we really need an auto-incrementing step_id? Won't a timestamp work and avoid tracking it? Will the step_id be guaranteed to be unique without locking?